### DISCUSSION TITLE
Is there a way to modify the request method in hono middleware?

### DISCUSSION THREAD

**Discussion by spencerbart** [honojs/hono#1089] [Q&A] [answered]
I am implementing the tus protocol and I need to be able to modify the incoming request method. I could just build a switch statement and do routing myself but then it reduces the need for hono. There's probably no way around this.

Is there any way to modify the request method in hono middleware?

![Screenshot 2023-05-10 at 15 24 11](https://github.com/honojs/hono/assets/38776747/0cc3b2ad-3dca-432d-806c-84f20304ba74)

**Comment by yusukebe** [comment, marked as answer]
Hi @spencerbart 

I think this is not a recommended approach, though, you can try this:

```ts
app.use('/', async (c, next) => {
  const req = c.req.raw.clone()
  c.req.raw = new Request(req, {
    method: 'POST',
  })
  await next()
  // ...
})
```

**Comment by davidvanleeuwen** [comment]
@spencerbart did you figure out how to work with tus + hono? I'm assuming you've created a separate implementation from tus-node?
